engine/gamesys/src/gamesys/components/comp_gui.cpp

```diff
@@ -1211,24 +1329,29 @@ namespace dmGameSystem
         ro.m_SetBlendFactors = 1;
 
         // If we need new render constants
-        HComponentRenderConstants render_constants = gui_world->m_RenderConstants[ro_count];
+        HComponentRenderConstants emitter_render_constants = gui_world->m_RenderConstants[ro_count];
         if (first_emitter_render_data->m_RenderConstantsSize > 0)
         {
-            if (!render_constants)
+            if (!emitter_render_constants)
             {
-                render_constants = dmGameSystem::CreateRenderConstants();
-                gui_world->m_RenderConstants[ro_count] = render_constants;
+                emitter_render_constants = dmGameSystem::CreateRenderConstants();
+                gui_world->m_RenderConstants[ro_count] = emitter_render_constants;
             }
         }
 
         for (uint32_t i = 0; i < first_emitter_render_data->m_RenderConstantsSize; ++i)
         {
             dmParticle::RenderConstant* c = &first_emitter_render_data->m_RenderConstants[i];
-            dmGameSystem::SetRenderConstant(render_constants, c->m_NameHash, (dmVMath::Vector4*) &c->m_Value, c->m_IsMatrix4 ? 4 : 1);
+            dmGameSystem::SetRenderConstant(emitter_render_constants, c->m_NameHash, (dmVMath::Vector4*) &c->m_Value, c->m_IsMatrix4 ? 4 : 1);
         }
 
-        if (render_constants) {
-            dmGameSystem::EnableRenderObjectConstants(&ro, render_constants);
+        // NOTE: This means that if any of the particle FX emitters has render constants, they will take precedence over the render cosntants
+        //       that has been set on the node via go.set.
+        HComponentRenderConstants render_constants_to_use = emitter_render_constants ? emitter_render_constants : render_constants;
```

Comment:
I _think_ this is correct? If someone used particlefx.set_constant on a specific emitter, it should probably take precedence over someone who has set the constant on a node via gui.set 🤔